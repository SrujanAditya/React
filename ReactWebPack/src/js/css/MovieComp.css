.movieBlock {
    display: inline-block;
    padding:20px;
    width:300px;
    margin:auto
}
.movieBlock:hover {
    box-shadow: 0 16px 32px 0 rgba(0,0,0,0.3)
}
.labelContainer {
    display:inline-block;
    width:250px
}
.movieTitle {
    font-weight:700;
    display:block
}
.movieGenres {
    font-weight:200;
    font-size:12px;
    display:block
}
.releaseYear {
    border-style:double;
    float:right;
    border-radius:5px;
    padding:2px
}