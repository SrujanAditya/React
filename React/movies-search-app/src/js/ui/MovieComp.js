import React from 'react';
import '../css/MovieComp.css';

const MovieComp = ({poster_path,id,title,genres,release_date,onClick,history}) => (
    <div className = "movieBlock" 
        onClick = {() => {
                    history.push("/movie/"+id);
                    onClick(id);
                }
            }>
        <img style={{width:'300px',borderRadius:'8px'}} src={poster_path} alt={title}/> 
        <div className="labelContainer">
            <label className="movieTitle">{title}</label>
            <label className="movieGenres">{genres.join(" & ")}</label>
        </div>
        <label className="releaseYear">{release_date.split("-")[0]}</label>
    </div>
)

export default MovieComp;