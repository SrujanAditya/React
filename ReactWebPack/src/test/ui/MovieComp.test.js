import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MovieComp from '../../js/ui/MovieComp';

Enzyme.configure({ adapter: new Adapter() })

describe('ui', () => {
    it('MovieComp component renders correctly', () => {
        const props = {
            poster_path:'',
            id:'',
            title:'',
            genres:['a','b'],
            release_date:'',
            onClick: jest.fn(),
            history : {}
        }
        const movieListComp = shallow(<MovieComp {...props} />);
        expect(movieListComp).toMatchSnapshot();
    }) 
})

describe('ui', () => {
    it('MovieComp test history push on click', () => {
        const props = {
            poster_path:'',
            id:'',
            title:'',
            genres:['a','b'],
            release_date:'',
            onClick:jest.fn()
        }
        const historyMock = { push : jest.fn(()=>console.log("CLICKED : History push")) };
        const wrapper= shallow(<MovieComp history={historyMock} {...props} />) 
        
        wrapper.find(".movieBlock").simulate("click");

        expect(historyMock.push.mock.calls.length).toBe(1);
    }) 
})
