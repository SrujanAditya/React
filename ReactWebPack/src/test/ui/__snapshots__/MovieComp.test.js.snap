// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui MovieComp component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieComp
    genres={
      Array [
        "a",
        "b",
      ]
    }
    history={Object {}}
    id=""
    onClick={[MockFunction]}
    poster_path=""
    release_date=""
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          src=""
          style={
            Object {
              "borderRadius": "8px",
              "width": "300px",
            }
          }
        />,
        <div
          className="labelContainer"
        >
          <label
            className="movieTitle"
          >
            
          </label>
          <label
            className="movieGenres"
          >
            a & b
          </label>
        </div>,
        <label
          className="releaseYear"
        >
          
        </label>,
      ],
      "className": "movieBlock",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "",
          "style": Object {
            "borderRadius": "8px",
            "width": "300px",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="movieTitle"
            >
              
            </label>,
            <label
              className="movieGenres"
            >
              a & b
            </label>,
          ],
          "className": "labelContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "movieTitle",
            },
            "ref": null,
            "rendered": "",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "a & b",
              "className": "movieGenres",
            },
            "ref": null,
            "rendered": "a & b",
            "type": "label",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "releaseYear",
        },
        "ref": null,
        "rendered": "",
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            src=""
            style={
              Object {
                "borderRadius": "8px",
                "width": "300px",
              }
            }
          />,
          <div
            className="labelContainer"
          >
            <label
              className="movieTitle"
            >
              
            </label>
            <label
              className="movieGenres"
            >
              a & b
            </label>
          </div>,
          <label
            className="releaseYear"
          >
            
          </label>,
        ],
        "className": "movieBlock",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "",
            "style": Object {
              "borderRadius": "8px",
              "width": "300px",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="movieTitle"
              >
                
              </label>,
              <label
                className="movieGenres"
              >
                a & b
              </label>,
            ],
            "className": "labelContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "movieTitle",
              },
              "ref": null,
              "rendered": "",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "a & b",
                "className": "movieGenres",
              },
              "ref": null,
              "rendered": "a & b",
              "type": "label",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "releaseYear",
          },
          "ref": null,
          "rendered": "",
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
